rust_src/src/windows.rs

```diff
@@ -622,4 +622,96 @@ pub extern "C" fn CURRENT_MODE_LINE_HEIGHT(mut window: LispWindowRef) -> i32 {
     window.current_mode_line_height()
 }
 
+/// Return a list of windows on FRAME, starting with WINDOW.
+/// FRAME nil or omitted means use the selected frame.
+/// WINDOW nil or omitted means use the window selected within FRAME.
+/// MINIBUF t means include the minibuffer window, even if it isn't active.
+/// MINIBUF nil or omitted means include the minibuffer window only
+/// if it's active.
+/// MINIBUF neither nil nor t means never include the minibuffer window.
+#[lisp_fn(min = "0")]
+pub fn window_list(
+    frame: Option<LispFrameRef>,
+    minibuf: LispObject,
+    window: Option<LispWindowRef>,
+) -> LispObject {
+    let w_obj = if window.is_none() {
+        if let Some(f) = frame {
+            f.selected_window()
+        } else {
+            selected_window()
+        }
+    } else {
+        window.unwrap().as_lisp_obj()
+    };
+
+    let f_obj: LispObject;
+    if let Some(f) = frame {
+        f_obj = f.as_lisp_obj();
+    } else {
+        f_obj = selected_frame();
+    }
+
+    let w_ref = w_obj
+        .as_window()
+        .unwrap_or_else(|| panic!("Invalid window reference."));
+
+    if !f_obj.eq(w_ref.frame()) {
+        error!("Window is on a different frame");
+    }
+
+    unsafe {
+        LispObject::from_raw(window_list_1(
+            w_obj.to_raw(),
+            minibuf.to_raw(),
+            f_obj.to_raw(),
+        ))
+    }
+}
+
```

Comment:
What is with this function? You are not using it.

Reply:
`window_list_one` ? It's lisp function ?

Reply:
I missed the name in the `lisp_fn` macro. Never mind.